lib/glimesh/chat.ex

```diff
@@ -365,10 +388,21 @@ defmodule Glimesh.Chat do
     Repo.update_all(query, set: [is_visible: false])
   end
 
+  defp delete_chat_message(%Channel{} = channel, message_id) do
+    query =
+      from m in ChatMessage,
+        where:
+          m.is_visible == true and
+            m.channel_id == ^channel.id and
+            m.id == ^message_id
```

Comment:
I suggest refactoring this function to take in a %ChatMessage{} object alone, and then you can simplify it to:
```
chat_message
|> Repo.update(set: [is_visible: false])
```

a ChatMessage object has everything it needs to do this query, and is unique across all channels because of it's id.